### ISSUE TITLE
add   "strictPropertyInitialization": false to tsconfig

### ISSUE THREAD

**Issue by ianhi** [jupyter-widgets/widget-ts-cookiecutter#74]
Given that with backbonejs you're meant to use initialize over the constructor this is a handy change. For example these lines in ipycanvas need require that setting.

https://github.com/martinRenou/ipycanvas/blob/09ca4e04c3c43bca5df46975cad28ada3d50750e/src/widget.ts#L66-L67

I was trying to do a similar thing and spent a while being mystified - so for someone making widgets without much typescript experience having this preset could be very nice.

Is there any reason not have this as a default?

**Comment by martinRenou** [comment]
I am a bit biased, but I am in favor for this :+1:. It's indeed very nice to be able to initialize attributes in the `initialize` method instead of overwriting the constructor.

**Comment by vidartf** [comment]
Closed via #75. On a side note, you will also need to be careful about the interaction between "class level initializers" (which @jasongrout prefer) vs initializing in `initialize`. I was just today working on a bug with code like this:

```ts
class Base(Widget):

  my_attribute: any[] = []


class Sub(Base):

  initialize(args, opts):
    super.initialize(args, opts);
    this.my_attribute = ['foo']

const sub = new Sub();
```

In the above code, what will `sub.my_attribute` be? Answer: `[]`, since the base level initializers will be added to the end of `Base`'s constructor, which will then execute after `initialize` has finished.

**Comment by martinRenou** [comment]
Good to know
